### DISCUSSION TITLE
adjusting angled x-axis label positions

### DISCUSSION THREAD

**Discussion by fred-spaceape** [JetBrains/lets-plot#1221] [Q&A] [answered]
I have a column chart (geom_bar) with a discrete x-axis of m/c dates. I've angled the labels by 45 degrees, but now they don't quite line up nicely. Ideally I'd like the month part to be lined up with the tick mark (right now the tick lines up with the middle of the rotated label).
I think in R ggplot you'd use hjust to fix the justification, but it doesn't seem to do anything when I invoke it in lets-plot.

`theme(axis_text_x=element_text(angle=45, hjust=0))`

<img width="739" alt="image" src="https://github.com/user-attachments/assets/af6b4eac-de04-4ef1-8f5e-d39670c30034">

**Comment by alshan** [comment]
Hi, unfortunately there is no good workaround in this situation. I've submitted an issue: https://github.com/JetBrains/lets-plot/issues/1227

**Comment by alshan** [comment, marked as answer]
Hi @fred-spaceape ! 
vjust,hjust are supported in [4.6.0](https://github.com/JetBrains/lets-plot?tab=readme-ov-file#what-is-new-in-460), see [demo](https://nbviewer.org/github/JetBrains/lets-plot/blob/master/docs/f-25a/axis_label_justification.ipynb).

**Reply by fred-spaceape** [reply]
OK I finally tried this, it worked great! Thanks for fixing :)

**Comment by fred-spaceape** [comment]
That’s great! I’ll give it a try

On Mon, 3 Mar 2025 at 21:27, Igor Alshannikov ***@***.***>
wrote:

> Hi @fred-spaceape <https://github.com/fred-spaceape> !
> vjust,hjust are supported in 4.6.0
> <https://github.com/JetBrains/lets-plot?tab=readme-ov-file#what-is-new-in-460>,
> see demo
> <https://nbviewer.org/github/JetBrains/lets-plot/blob/master/docs/f-25a/axis_label_justification.ipynb>
> .
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JetBrains/lets-plot/discussions/1221#discussioncomment-12380972>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACWABX645ZKFM2L4FB7O4Y32STCL7AVCNFSM6AAAAABQL6PRI2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTEMZYGA4TOMQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 













We're hiring! Go to spaceapegames.com 
<http://www.spaceapegames.com> for more!
Space Ape Games (UK) Limited is 
registered in England and Wales, No. 7956131. 
Registered office: First 
Floor, 100 New Oxford Street, London WC1A 1HB.
All communication is dealt 
in line with our Privacy Policy 
<http://www.spaceapegames.com/privacy-policy> and Recruitment Privacy 
Policy <http://www.spaceapegames.com/candidate-privacy-policy>.
